### PR TITLE
WelcomScreenBase

### PR DESCRIPTION
**Pull request by soniaigartua**
I developed the WelcomeScreem, using the structure MVP. Then, I added the Unit Test.

Evidence:
app

Tests:
Anotación 2019-07-26 114418

### CODE REVIEW

File: app/src/main/java/com/example/pps_tudai/mvp/presenter/MainPresenter.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    public void onLoginPressed() {
+        mainView.showWelcomeMessage();
+    }
+
+    public void onSignUpPressed() {
+        mainView.showRegistrationScreen();
+    }
+
+
+//    @Override
```
Comment (SartiMau) [inline app/src/main/java/com/example/pps_tudai/mvp/presenter/MainPresenter.java]: remove commented code pls

File: app/src/main/java/com/example/pps_tudai/mvp/view/MainView.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    private WeakReference<Activity> activityWeak;
+
+    public MainView(Activity activity) {
+        ButterKnife.bind(this, activity);
+        this.activityWeak = new WeakReference<Activity>(activity);
+    }
+
+    public void showWelcomeMessage() {
+        if (activityWeak.get() != null) {
+            Toast.makeText(activityWeak.get(), "Welcome to the app", Toast.LENGTH_LONG).show();
```
Comment (SartiMau) [inline app/src/main/java/com/example/pps_tudai/mvp/view/MainView.java]: no hardcode pls.. here you can use a string resource and access it by R.string.YOUR_STRING_NAME

File: app/src/main/java/com/example/pps_tudai/mvp/view/MainView.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        }
+    }
+
+    public void showRegistrationScreen() {
+        if (activityWeak.get() != null) {
+            Toast.makeText(activityWeak.get(), "Complete the next form", Toast.LENGTH_LONG).show();
```
Comment (SartiMau) [inline app/src/main/java/com/example/pps_tudai/mvp/view/MainView.java]: same

File: app/src/main/res/layout/activity_main.xml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".activity.MainActivity">
+
+    <Button
+        android:id="@+id/btn_login"
+        android:layout_width="match_parent"
+        android:layout_height="101dp"
```
Comment (SartiMau) [inline app/src/main/res/layout/activity_main.xml]: no hardcode pls use dimens

File: app/src/main/java/com/example/pps_tudai/mvp/presenter/MainPresenter.java (hunk 1 above)
Comment (soniaigartua) [reply app/src/main/java/com/example/pps_tudai/mvp/presenter/MainPresenter.java]: done!

File: app/src/main/java/com/example/pps_tudai/mvp/view/MainView.java (hunk 2 above)
Comment (soniaigartua) [reply app/src/main/java/com/example/pps_tudai/mvp/view/MainView.java]: done!

File: app/src/main/java/com/example/pps_tudai/mvp/view/MainView.java (hunk 3 above)
Comment (soniaigartua) [reply app/src/main/java/com/example/pps_tudai/mvp/view/MainView.java]: done!

File: app/src/main/res/layout/activity_main.xml (hunk 4 above)
Comment (soniaigartua) [reply app/src/main/res/layout/activity_main.xml]: done! ... I changed the way of customize the buttons from the layout

**Pull request merged**